api/src/services/assets.ts

```diff
@@ -66,9 +66,48 @@ export class AssetsService {
 		if (!exists) throw new ForbiddenException();
 
 		if (range) {
-			if (range.start >= file.filesize || (range.end && range.end >= file.filesize)) {
+			if (!range.start && !range.end) {
 				throw new RangeNotSatisfiableException(range);
 			}
+
```

Comment:
It's easier to read these branches when the control statements occur up-front. Some semantic case names would improve the legibility of bad cases too:
```javascript
if (range) {
  const missingRangeLimits = !range.start && !range.end;
  const endBeforeStart = range.start && range.end && range.end <= range.start
  const startOverflow = range.start && range.start >= file.filesize;
  const endUnderflow = range.end && range.end <= 0;

  if ( missingRangeLimits || endBeforeStart || startOverflow || endUnderflow ) {
    throw new RangeNotSatisfiableException(range);
  }

  const lastByte = file.filesize - 1;

  if (range.end) {
    if (!range.start) {
      // fetch chunk from tail
      range.start = file.filesize - range.end;
      range.end = lastByte;
    }

    if (range.end >= file.filesize) {
      // fetch entire file
      range.end = lastByte;
    }
  }

  if (range.start) {
    if (!range.end) {
      // fetch entire file
      range.end = lastByte;
    }

    if (range.start < 0) {
      // fetch file from head
      range.start = 0;
    }
  }
}
```

Reply:
Implemented 👍